Source/Common/Profiler.h

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+/*
+$info$
+tags: Common|Profiler
+desc: Frontend profiler common code
+$end_info$
+*/
+#pragma once
+#include <FEXCore/Utils/Profiler.h>
+
+namespace FEXCore::Core {
+struct InternalThreadState;
+}
+
+#ifdef _M_ARM_64
+static inline void store_memory_barrier() {
+  asm volatile("dmb ishst;" ::: "memory");
+}
+
+#else
+static inline void store_memory_barrier() {
+  // Intentionally empty.
+  // x86 is strongly memory ordered with regular loadstores. No need for barrier.
+}
+#endif
+
+namespace FEX::Profiler {
+class StatAllocBase {
+protected:
+  FEXCore::Profiler::ThreadStats* AllocateSlot(uint32_t TID);
+  void DeallocateSlot(FEXCore::Profiler::ThreadStats* AllocatedSlot);
+
+  uint32_t OffsetFromStat(FEXCore::Profiler::ThreadStats* Stat) const {
+    return reinterpret_cast<uint64_t>(Stat) - reinterpret_cast<uint64_t>(Base);
+  }
+  uint32_t TotalSlotsFromSize() const {
+    return (CurrentSize - sizeof(FEXCore::Profiler::ThreadStatsHeader)) / sizeof(FEXCore::Profiler::ThreadStats) - 1;
+  }
+  uint32_t TotalSlotsFromSize(uint32_t Size) const {
+    return (Size - sizeof(FEXCore::Profiler::ThreadStatsHeader)) / sizeof(FEXCore::Profiler::ThreadStats) - 1;
+  }
+
+  uint32_t SlotIndexFromOffset(uint32_t Offset) {
```

Comment:
Should this be const or static, like the above function?

Reply:
Good catch, both this and the above function can be static. Switched over.